class-based/src/step-function.ts

```diff
@@ -0,0 +1,16 @@
+import { Construct } from "constructs";
+import { StepFunction } from "functionless";
+import { SayFunction } from "./say-function";
+
+interface SayFunctionProps {
+  sayFunction: SayFunction["fn"];
+}
+
+export class Workflow extends Construct {
+  constructor(scope: Construct, id: string, props: SayFunctionProps) {
+    super(scope, id);
+    new StepFunction(scope, "Workflow", async (event: { name: string }) => {
+      await props.sayFunction({ message: `Hello ${event.name}` });
+    });
+  }
+}
```

Comment:
There is another option here when the step function is sub-classed rather than wrapped. This would get rid of the naming issue.

```ts
export class Workflow extends StepFunction {
   constructor(...) {
      super(scope, id, async () => {

      });
   }
}
```

Reply:
If I was encapsulating logic I would prefer this approach or the functional approach.

In the case of needing multiple interconnected resource, the construct makes the most sense.

Reply:
I had a third variation at one point that had this as well. I think the number of way there are to slice this is a part of the reason people complain about CDK being too open.